dependencies.py

```diff
@@ -67,7 +67,7 @@ def extra_requirements() -> dict[str, list[str]]:
         "redis": ["redis~=4.3"],
         "mlflow": ["mlflow~=2.16"],
         "databricks-sdk": ["databricks-sdk~=0.20.0"],
-        "sqlalchemy": ["sqlalchemy~=1.4"],
+        "sqlalchemy": ["sqlalchemy~=2.0"],
```

Comment:
@assaf758 please approve the change

Reply:
LGTM. BTW, SQLTarget / SQLSource is tech-preview (never tested by QA) and is not in use.